model/src/main/kotlin/licenses/LicenseInfoResolver.kt

```diff
@@ -203,28 +206,47 @@ class LicenseInfoResolver(
 
         val licenseInfo = resolveLicenseInfo(id)
         val licenseFiles = mutableListOf<ResolvedLicenseFile>()
-        val matcher = FileMatcher(archiver.patterns)
 
         licenseInfo.flatMapTo(mutableSetOf()) { resolvedLicense ->
             resolvedLicense.locations.map { it.provenance }
         }.forEach { provenance ->
             val archiveDir = createTempDirectory("$ORT_NAME-archive").toFile().apply { deleteOnExit() }
-
             val path = "${id.toPath()}/${provenance.hash()}"
-            if (archiver.unarchive(archiveDir, path)) {
-                archiveDir.walk().forEach { file ->
-                    val relativePath = file.toRelativeString(archiveDir)
-                    if (matcher.matches(relativePath)) {
-                        licenseFiles += ResolvedLicenseFile(
-                            provenance, licenseInfo.filter(provenance, relativePath), relativePath, file
-                        )
-                    }
-                }
+
+            if (!archiver.unarchive(archiveDir, path)) return@forEach
+
+            val rootLicenseFiles = getApplicableLicenseFilesForDirectory(
+                relativeFilePaths = archiveDir.walk().filter { it.isFile }.mapTo(mutableSetOf()) {
+                    it.relativeTo(archiveDir).toString()
```

Comment:
`it.toRelativeString(archiveDir)` instead?

Reply:
Done.